examples/example-gateway/endpoints/baz/baz_simpleservice_method_call_test.go

```diff
@@ -63,7 +63,7 @@ func TestCallSuccessfulRequestOKResponse(t *testing.T) {
 	}
 	defer gateway.Close()
 
-	gateway.TChannelBackends()["baz"].Register(bazServer.NewServerWithSimpleServiceCall(call))
+	gateway.TChannelBackends()["baz"].Register(bazServer.NewSimpleServiceCallHandler(call))
```

Comment:
Somehow this feels wrong.

`Register({ Service, Method, RequestHandler })` feels wrong compared to `Register("Service::Method", RequestHandler)`

It's weird that we register an interface that contains methods to get the arg1 name.

Definitely not symmetrical with registering http endpoints